### PR TITLE
feat: shielded sync with multicore

### PR DESCRIPTION
**Pull request by mateuszjasiuk**
Adds multicore support for shielded sync.

To test:
- run `yarn wasm:build:dev:multicore`
 - add this in App.tsx
```ts
// You can import worker like this
// import ShieldedSyncWorker from "workers/ShieldedSyncWorker?worker";
// ---

  const worker = new ShieldedSyncWorker();
  const rpcUrl = useAtomValue(rpcUrlAtom);
  const maspIndexerUrl = useAtomValue(maspIndexerUrlAtom);

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  (window as any).shieldedSync = () => {
    const msg: ShieldedSyncMessageType = {
      type: "shielded-sync-multicore",
      payload: {
        rpcUrl,
        maspIndexerUrl,
        vks: [
          "zvknam1...",
        ],
      },
    };

    worker.postMessage(msg);
  };
```
You can also test singlecore by running `yarn wasm:buid:dev` and changing type to `"shielded-sync-singlecore"`
**or**
You can test both singlecore and multicore without the MASP Indexer, just remove `maspIndexerUrl`

Seems to work fine for FF too, it is slower though.

### CODE REVIEW

File: apps/namadillo/src/App/worker.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import init, { initMulticore } from "@heliax/namada-sdk/inline-init";
```
Comment (mateuszjasiuk) [inline apps/namadillo/src/App/worker.ts:1]: rename worker.ts and move to workers folder or sth

File: apps/namadillo/src/App/App.tsx [hunk 2]
```diff
@@ -16,6 +20,26 @@ export function App(): JSX.Element {
   useTransactionNotifications();
   useTransactionCallback();
 
+  const worker = new Worker();
```
Comment (mateuszjasiuk) [inline apps/namadillo/src/App/App.tsx:23]: cleanup this stuff

File: apps/namadillo/src/workers/ShieldedSyncWorker.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  | ShieldedSyncSinglecore;
+
+self.onmessage = async (e: MessageEvent<ShieldedSyncMessageType>) => {
+  const { type, payload } = e.data;
+
+  switch (type) {
+    case "shielded-sync-singlecore": {
+      const { cryptoMemory } = await init();
+      //eslint-disable-next-line
+      console.log("Syncing with single core");
```
Comment (jurevans) [inline apps/namadillo/src/workers/ShieldedSyncWorker.ts:32]: Maybe we could change these to `console.info` to avoid the eslint-disable? That might work

**Review by jurevans — APPROVED** [review]
Nice, this worked perfectly for me in Chrome & Firefox!

**Pull request merged**